Frontend/app/class/detail/detail.component.ts

```diff
@@ -66,11 +84,153 @@ export class ClassDetailComponent implements OnInit, OnDestroy {
     this.isLoading = false;
   }
 
-  handleError(err){
-    this.toastr.error("Unable to access class.");
+  handleError(){
+    this.toastr.error('Unable to access class.');
     this.router.navigateByUrl('/dashboard');
   }
 
+  /* Activity Modal Methods */
+  showActivityModal():void {
+    this.newActivity = new ActivityModel('', undefined, 5, undefined, undefined, undefined, undefined, undefined);
```

Comment:
is there a reason you are using undefined instead of just ` ' ' ` for strings in the model?